### DISCUSSION TITLE
Add configurable poll interval for SDS011

### DISCUSSION THREAD

**Discussion by robin13** [arendst/Tasmota#13420] [Q&A] [answered]
The SDS011 air quality (fine dust) sensor uses lasers.
From the data sheet:

> Service life is the key parameter of laser dust sensor. The laser diode in this sensor has high quality and its service life is up to 8000 hours

8000 hours continuous use can add up quickly: about 1.5 years in continuous use.
Currently Tasmota polls every second, and I _think_ the device never goes to sleep:
https://github.com/arendst/Tasmota/blob/5e805e0c4ac1304bfe86a3abd1b6b99b7dda2daa/tasmota/xsns_20_novasds.ino#L147

Would it be possible to have a configurable poll interval so that depending on usecase the device could power up LED and fan, take a reading, and then sleep for a minute; thereby extending the life of the LED considerably?

**Comment by sfromis** [comment, marked as answer]
There is already such a feature, using the command [Sensor20](https://tasmota.github.io/docs/Commands/#sensor20) in association with setting TelePeriod for how often you want readings.

BTW, the lifetime which is being extended is AFAIK not the LED, but to prevent the sensor innards getting dirty from the air sucked in.

**Comment by robin13** [comment]
Oooh... thanks for the pointer, and also for updating the docs:
https://github.com/tasmota/docs/commit/8a48f530b76af23f517fe87121e5912a57767a6f!
I have now set
```
Sensor20 5
TelePeriod 300
```
I see the default value for Sensor20 is `30` ... is 30s really necessary, or is this to take account for perhaps the length of a tube attached to the sensor, and the necessary time to pull a sample of "outside" air into the sample chamber?

**Reply by barbudor** [reply]
The sensor needs to heat up (literally) so you need such time to reach the proper temperature for correct measurement

**Reply by sfromis** [reply]
According to "Laser Dust Sensor Control Protocol V1.3", around sleeping mode.
> Notes: The data is stable when the sensor works after 30 seconds;
> The fan and laser stop working in sleeping mode

**Reply by robin13** [reply]
Ah - good, then I'll do
```
Sensor20 30
TelePeriod 300
```
:+1:
